src/util.ts

```diff
@@ -255,4 +259,68 @@ export interface SynthesizedApp {
    * The construct metadata file (if exists).
    */
   readonly constructMetadata?: string;
+}
+
+export function parseImports(spec: string): ImportSpec {
+  const splitImport = spec.split(PREFIX_DELIM);
+
+  // k8s@x.y.z
+  // crd.yaml
+  // url.com/crd.yaml
+  if (splitImport.length === 1) {
+    return {
+      source: spec,
+    };
+  }
+
+  // crd:=crd.yaml
+  // crd:=url.com/crd.yaml
+  if (splitImport.length === 2) {
+    return {
+      moduleNamePrefix: splitImport[0],
+      source: splitImport[1],
+    };
+  }
+
+  throw new Error('Unable to parse import specification. Syntax is [NAME:=]SPEC');
+}
+
+export function deriveFileName(url: string) {
+  const devUrl = matchCrdsDevUrl(url);
+  let filename = undefined;
+
+  if (devUrl) {
+    const lastIndexOfSlash = devUrl.lastIndexOf('/');
+    const lastIndexOfAt = devUrl.lastIndexOf('@');
+    filename = devUrl.slice(lastIndexOfSlash+1, lastIndexOfAt);
+  } else {
+    const lastIndexOfSlash = url.lastIndexOf('/');
+    const lastIndexOfYaml = url.lastIndexOf('.yaml');
+    filename = url.slice(lastIndexOfSlash+1, lastIndexOfYaml);
+  }
+
+  if (!filename) {
+    filename = createHash('sha256');
+  }
+
+  return filename;
+}
+
+export function isK8sImport(value: string) {
+  if (value !== 'k8s' && !value.startsWith('k8s@')) {
+    return false;
+  }
+
+  return true;
+}
+
+export function crdsArePresent(imprts: string[] | undefined) {
+  if (!imprts) {
+    return false;
+  }
+  if (imprts.length === 0 || (imprts.length === 1 && isK8sImport(imprts[0]))) {
+    return false;
+  }
+
+  return true;
```

Comment:
This would return true for:

```yaml
- k8s@1.22
- k8s@1.23
```

Which is incorrect.

```suggestion
return (imprts ?? []).some(i => !isK8sImport(i));
```